auth/gcloud/aio/auth/token.py

```diff
@@ -132,12 +143,20 @@ def get_service_data(
         return {}
 
 
-class Token:
+@dataclass
+class TokenResponse:
+    value: str
+    expires_in: int = GCLOUD_TOKEN_DURATION
+
+
+class BaseToken:
+    """GCP auth token base class."""
     # pylint: disable=too-many-instance-attributes
+    __metaclass__ = ABCMeta
```

Comment:
Do you have any good references on __metaclass__ and working with metaclasses in Python? Interest only.

Reply:
The only place that has served me well on this has been the official Python docs unfortunately! I'm happy to field questions if you have specific ones :)

https://docs.python.org/3/library/abc.html